### ISSUE TITLE
Unable to use roboflow's dataset for training

### ISSUE THREAD

**Issue by qq996235826** [ultralytics/ultralytics#1664]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Bug

I used the dataset on roboflow for training, and it was successful the first few times, but when I remove a label, YOLOV8 seems to recognize my dataset as object detection.
All you need to do to reproduce the BUG is train with my data set
this is my [dataset](https://app.roboflow.com/jiangsu-university-of-science-and-technology/yolov8-lefuqiu/13/export)
![227894812-db23df79-3710-4b83-b53b-e23a94350d54](https://user-images.githubusercontent.com/51869146/228109089-b01f1890-af57-4fcb-b644-907e01d0b933.png)

### Environment

yolo：ultralytics   8.0.58       torch  2.0.0+cu118
python:3.9
OS:WIN11

### Minimal Reproducible Example

```
import os
os.environ["KMP_DUPLICATE_LIB_OK"]="TRUE"
from ultralytics import YOLO
import cv2 as cv

if __name__ == '__main__':
    # Load a model
    model = YOLO('yolov8n-seg.yaml')   # build a new model from scratch

    # Train the model
    model.train(data="datasets/data.yaml", epochs=200, imgsz=640, device=0)
```

### Additional

I didn't change the YOLO code
Only the data downloaded from roboflow was used for training

### Are you willing to submit a PR?

- [X] Yes I'd like to help by submitting a PR!

**Comment by qq996235826** [comment]
If you use the [previous version](https://app.roboflow.com/jiangsu-university-of-science-and-technology/yolov8-lefuqiu/8), the training will be fine
But if you choose to drop the [catch class](https://app.roboflow.com/jiangsu-university-of-science-and-technology/yolov8-lefuqiu/12), the exported data set becomes untrainable
Here is my directory structure
![屏幕截图 2023-03-28 101404](https://user-images.githubusercontent.com/51869146/228109833-13bcf5d0-2e51-440e-ac46-cc2448114449.png)

**Comment by AyushExel** [comment]
This is the expected behavior. For the segmentation dataset, the length of boxes and segment labels should be equal. If they are not equal, the dataset will be treated as a detection dataset.

**Comment by pderrenger** [comment]
@qq996235826 you're welcome! Feel free to reach out if you have any more questions or need further assistance with your dataset. Good luck with your debugging!

**Comment by glenn-jocher** [comment]
@pderrenger you're welcome! If you need further assistance, don't hesitate to ask. Good luck with your dataset! 👍
